lbrynet/core/Wallet.py

```diff
@@ -130,54 +236,161 @@ def create_tables(transaction):
                                 "    name text, " +
                                 "    txid text, " +
                                 "    n integer)")
+            transaction.execute("CREATE TABLE IF NOT EXISTS claim_cache (" +
+                                "    row_id INTEGER PRIMARY KEY AUTOINCREMENT, " +
+                                "    claim_id TEXT UNIQUE NOT NULL, " +
+                                "    claim_sequence INTEGER, " +
+                                "    claim_address TEXT NOT NULL, " +
+                                "    height INTEGER NOT NULL, " +
+                                "    amount INTEGER NOT NULL, " +
+                                "    supports TEXT, " +
+                                "    claim_pb TEXT, " +
+                                "    channel_name TEXT, " +
+                                "    signature_is_valid BOOL, " +
+                                "    last_modified TEXT)")
+            transaction.execute("CREATE TABLE IF NOT EXISTS uri_cache (" +
+                                "    row_id INTEGER PRIMARY KEY AUTOINCREMENT, " +
+                                "    uri TEXT, " +
+                                "    cache_row INTEGER, " +
+                                "    certificate_row INTEGER, " +
+                                "    last_modified TEXT)")
 
         return self.db.runInteraction(create_tables)
 
+    @rerun_if_locked
+    @defer.inlineCallbacks
     def clean_bad_records(self):
-        d = self.db.runQuery("delete from name_metadata where length(txid) > 64 or txid is null")
-        return d
+        yield self.db.runQuery("DELETE FROM name_metadata WHERE LENGTH(txid) > 64 OR txid IS NULL")
+        defer.returnValue(None)
 
+    @rerun_if_locked
+    @defer.inlineCallbacks
     def save_name_metadata(self, name, claim_outpoint, sd_hash):
-        d = self.db.runQuery(
-            "delete from name_metadata where name=? and txid=? and n=? and sd_hash=?",
-            (name, claim_outpoint['txid'], claim_outpoint['nout'], sd_hash))
-        d.addCallback(
-            lambda _: self.db.runQuery(
-                "delete from name_metadata where name=? and txid=? and n=? and sd_hash=?",
-                (name, claim_outpoint['txid'], UNSET_NOUT, sd_hash)))
-        d.addCallback(
-            lambda _: self.db.runQuery(
-                "insert into name_metadata values (?, ?, ?, ?)",
-                (name, claim_outpoint['txid'], claim_outpoint['nout'], sd_hash)))
-        return d
+        txid, nout = claim_outpoint['txid'], claim_outpoint['nout']
+        record_exists = yield self.db.runQuery("SELECT COUNT(*) FROM name_metadata "
+                                               "WHERE name=? AND txid=? AND n=?",
+                                               (name, txid, nout))
+        if not record_exists:
+            yield self.db.runOperation("INSERT INTO name_metadata VALUES (?, ?, ?, ?)",
```

Comment:
if this can't be one query for now, at least add a `TODO` line so we know to come back to it later